src/Button/index.js

```diff
@@ -1,79 +1,215 @@
+  if (props.isOutlined) {
+    return "transparent";
+  } else if (props.color !== "primary" && props.color !== "secondary") {
+    return props.color;
+  } else {
+    return theme.colors[props.color];
+  }
+}
+
+function getHoverColor(theme, props) {
+  if (
+    props.isOutlined &&
+    props.color !== "primary" &&
+    props.color !== "secondary"
+  ) {
+    return Color(props.color)
+      .lighten(0.55)
+      .string();
+  } else if (
+    !props.isOutlined &&
+    props.color !== "primary" &&
+    props.color !== "secondary"
+  ) {
+    return Color(props.color)
+      .darken(0.1)
+      .string();
+  } else if (props.isOutlined) {
+    return Color(theme.colors[props.color])
+      .lighten(0.55)
+      .string();
+  } else {
+    return Color(theme.colors[props.color])
+      .darken(0.1)
+      .string();
+  }
+}
+
+function getRippleColor(theme, props) {
+  if (
+    props.isOutlined &&
+    props.color !== "primary" &&
+    props.color !== "secondary"
+  ) {
+    return Color(props.color)
+      .lighten(0.45)
+      .string();
+  } else if (
+    !props.isOutlined &&
+    props.color !== "primary" &&
+    props.color !== "secondary"
+  ) {
+    return props.color;
+  } else if (props.isOutlined) {
+    return Color(theme.colors[props.color])
+      .lighten(0.45)
+      .string();
+  } else {
+    return theme.colors[props.color];
+  }
+}
+
+function getBorderColor(theme, props) {
+  if (props.color !== "primary" && props.color !== "secondary") {
+    return props.color;
+  } else {
+    return theme.colors[props.color];
+  }
+}
+
+class Button extends Component<Props, State> {
+  static defaultProps = {
+    element: "button",
+    color: "primary",
+    isOutlined: false,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      rippleActive: false,
+    };
+  }
+
+  toggleRipple = () => {
+    this.setState(
+      (state) => ({
+        rippleActive: !state.rippleActive,
+      }),
+      () => {
+        return setTimeout(() => {
+          this.setState((state) => ({
+            rippleActive: !state.rippleActive,
+          }));
+        }, 401);
+      }
+    );
+  };
+
+  handleClick = (e) => {
+    const button = e.target;
+
+    let parentOffset = button.getBoundingClientRect(),
+      relX = e.pageX - parentOffset.left,
+      relY = e.pageY - parentOffset.top;
+
+    const span = button.parentElement.getElementsByTagName("span");
+
+    span[0].style.top = relY + "px";
+    span[0].style.left = relX + "px";
+
+    this.toggleRipple();
+  };
+
+  render() {
+    const {
+      children,
+      classes,
+      className,
+      color,
+      element: Element,
+      icon,
+      innerClassName,
+      isOutlined,
+      theme,
+      ...rest
+    } = this.props;
+    const { rippleActive } = this.state;
+
+    return (
+      // $FlowFixMe
+      <Element className={classNames(classes.root, className)} {...rest}>
+        <div
+          onClick={this.handleClick}
```

Comment:
Can we put the onClick on the `Element` tag (which is almost always a `button` or `a` tag) rather than on the child div? This'll also require updating the `handleClick` method a bit... I'll leave a note there too.